src/nav/ChatNavBar.js

```diff
@@ -25,6 +25,9 @@ export default function ChatNavBar(props: Props) {
   const streamColor = useSelector(state => getStreamColorForNarrow(state, narrow));
   const color =
```

Comment:
As a small followup improvement:

Now that `color` isn't used for text anymore, it's just used for `<FooButton />`s, it'll be easier to give it a more specific name, which should help solidify its purpose. What do you think about `buttonColor`? Maybe something with "button," if we suppose the `<FooButton />`s are appropriately named, which I guess they are (though they make me think of `ZulipButton` which is quite different, hmm).

Reply:
And another potential small follow-up improvement:

I think there should be a nice way to avoid repeating the `streamColor === undefined` conditional, which might make things a bit more readable.

Is this it something like this:

```js
  const streamColor = useSelector(state => getStreamColorForNarrow(state, narrow));
  const themeColor = useContext(ThemeContext).color;

  const [buttonColor, textColor, spinnerColor] =
    streamColor === undefined // <----- (conditional happens just once, here)
      ? [BRAND_COLOR, themeColor, 'default']
      : [
          foregroundColorFromBackground(streamColor),
          foregroundColorFromBackground(streamColor),
          foregroundColorFromBackground(streamColor),
        ];
```

Or, could use object destructuring instead of array; it'd be more verbose because we'd repeat `buttonColor`, `textColor`, and `spinnerColor` a few times.

Reply:
I added the `color` → `buttonColor` commit to this, since that seems obviously better.

For the other thing, how would you feel about just moving `streamColor === undefined` into a named variable, and then keeping the three conditionals? That feels cleaner to me, but I'm open to being convinced otherwise.

Reply:
Yeah, I think that'd be good.